### PR TITLE
crns-112/auto complete input upgrade

### PR DESCRIPTION
**Pull request by virdesai**
# Submit a pull request

- Converting AutoCompleteInput components from class-based to functional components
- Utilizing SuggestionsContext and TranslationContext instead of prop drilling

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

### CODE REVIEW

**Comment by vishalnarkhede** [comment]
So far lgtm!!

File: src/components/AutoCompleteInput/AutoCompleteInput.js [hunk 1]
```diff
@@ -1,258 +1,232 @@
...
+  } = useContext(SuggestionsContext);
+  const { t } = useContext(TranslationContext);
+
+  const isTrackingStarted = useRef(false);
+
+  const [selectionEnd, setSelectionEnd] = useState(0);
+
+  useEffect(() => {
+    handleChange(value, true);
+  }, [handleChange, value]);
```
Comment (DanC5) [inline src/components/AutoCompleteInput/AutoCompleteInput.js:51]: does handleChange need to be in the dep array?

File: src/components/AutoCompleteInput/AutoCompleteInput.js [hunk 2]
```diff
@@ -1,258 +1,232 @@
...
+   * @see See https://facebook.github.io/react-native/docs/textinput#reference
+   */
+  additionalTextInputProps: PropTypes.object,
+  /**
+   * @param text string
+   */
+  onChange: PropTypes.func,
+  setInputBoxRef: PropTypes.func,
+  triggerSettings: PropTypes.object,
+  value: PropTypes.string,
```
Comment (DanC5) [inline src/components/AutoCompleteInput/AutoCompleteInput.js:229]: after our chat, prob wanna add descriptions to these guys

File: src/components/AutoCompleteInput/AutoCompleteInput.js (hunk 1 above)
Comment (virdesai) [reply src/components/AutoCompleteInput/AutoCompleteInput.js:51]: doesn't hurt

File: src/components/AutoCompleteInput/AutoCompleteInput.js (hunk 2 above)
Comment (virdesai) [reply src/components/AutoCompleteInput/AutoCompleteInput.js:229]: nah chill bro

File: src/components/AutoCompleteInput/__tests__/AutoCompleteInput.test.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    rerender(getComponent(props));
+
+    await waitFor(() => {
+      expect(queryByTestId('auto-complete-text-input')).toBeTruthy();
+      expect(props.closeSuggestions).toHaveBeenCalledTimes(1);
+      expect(props.openSuggestions).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  // TODO: figure out how to make tests work for @ mentions with needing to update function state values
```
Comment (DanC5) [inline src/components/AutoCompleteInput/__tests__/AutoCompleteInput.test.js:92]: the @ mention popup always displays the members of the channel, so couldn't we add members to our mock channel and ensure they show up when @ is typed?

Comment (virdesai) [reply src/components/AutoCompleteInput/__tests__/AutoCompleteInput.test.js:92]: the issue is that the native event handler for onSelectionChange doesn't trigger from these test updates through react-test-render and so the handleSuggestion method doesn't operate as it should and the tests don't operate correctly for it

File: src/components/AutoCompleteInput/AutoCompleteInput.js (hunk 2 above)
Comment (virdesai) [reply src/components/AutoCompleteInput/AutoCompleteInput.js:229]: I'll handle it within my MessageInput changes

**Pull request merged**